### PR TITLE
Add lowering of verify intrinsic.

### PR DESCRIPTION
**Pull request by mleair**
I was planning to include kind specialized versions in this pull request, but I hit a snag with them. So, I'm submitting the generalized runtime descriptor version for now. I'll seek help on the specialized version.

### CODE REVIEW

File: flang/include/flang/Lower/CharacterRuntime.h [hunk 1]
```diff
@@ -65,6 +65,13 @@ void genScan(Fortran::lower::FirOpBuilder &builder, mlir::Location loc,
 void genTrim(Fortran::lower::FirOpBuilder &builder, mlir::Location loc,
              mlir::Value resultBox, mlir::Value stringBox);
 
+/// Generate call to verify runtime.
+/// This calls the descriptor based runtime call implementation of the scan
```
Comment (jeanPerier) [inline flang/include/flang/Lower/CharacterRuntime.h:69]: `the scan` -> `the verify`

**Review by jeanPerier — APPROVED** [review]
Looks good to me

**Review by schweitzpgi — COMMENTED** [review]
Thanks, Mark. Just noting that the boxed calls are less efficient.

**Comment by klausler** [comment]
Are you going to be calling only the descriptor-based runtime APIs for all intrinsics?  If so, there's a lot of code that can be deleted from the runtime.  Let me know.

**Comment by mleair** [comment]
@klausler We will use both versions of the runtime. @schweitzpgi answered my question on when/how to call the specialized versions, so I plan to create calls to specialized versions of scan and verify soon.

**Pull request merged**